### ISSUE TITLE
XML parser problem handling fields marked as enum in common.xml

### ISSUE THREAD

**Issue by tstellanova** [tstellanova/rust-mavlink#1]
common.xml has multiple message types with fields marked as "enum" -- these are currently interpreted by the parser as being rust enum types, but they are not.  They are often flag fields where the "enum" values can be OR'd together on write (or tested on read). The parser-generated code panics while calling `FromPrimitive`.

cc  @podhrmic

Example: Parsing the `HIL_ACTUATOR_CONTROLS` message fails because the px4_sitl sender sets the "enum" mode as follows:

```
msg.mode = mode_flag_custom;
msg.mode |= (armed) ? mode_flag_armed : 0;
msg.flags = 0;
```

for reference, the `HIL_ACTUATOR_CONTROLS` msg from common.xml:
```
    <message id="93" name="HIL_ACTUATOR_CONTROLS">
      <description>Sent from autopilot to simulation. Hardware in the loop control outputs (replacement for HIL_CONTROLS)</description>
      <field type="uint64_t" name="time_usec" units="us">Timestamp (UNIX Epoch time or time since system boot). The receiving end can infer timestamp format (since 1.1.1970 or since system boot) by checking for the magnitude the number.</field>
      <field type="float[16]" name="controls">Control outputs -1 .. 1. Channel assignment depends on the simulated hardware.</field>
      <field type="uint8_t" name="mode" enum="MAV_MODE">System mode. Includes arming state.</field>
      <field type="uint64_t" name="flags" display="bitmask">Flags as bitfield, reserved for future use.</field>
    </message>
```

**Comment by tstellanova** [comment]
This was fixed by editing common.xml to redefine   HIL_ACTUATOR_CONTROLS mode field from
```
<field type="uint8_t" name="mode" enum="MAV_MODE">System mode. Includes arming state.</field>
```
to
```
<field type="uint8_t" name="mode" enum="MAV_MODE_FLAG"  display="bitmask">System mode. Includes arming state.</field>
```

I'm not sure how the common.xml file was originally generated.

**Comment by podhrmic** [comment]
I am using [bitflags](https://docs.rs/bitflags/1.0.4/bitflags/?search=) for the enum generation. You should be able to use [from_bits](https://docs.rs/bitflags/1.0.4/bitflags/example_generated/struct.Flags.html#method.from_bits) (or `from_bits_truncate`) to create an enum from the bitfields. It gets tricky when the enum has values that are not specified in the common.xml - like some secret or user defined values. But that can be fixed by enhancing the xml file with additional enum variants (and hopefully these changes can be upstreamed)

`Common.xml`is a copy of the common.xml from Mavlink message definition [here](https://docs.rs/bitflags/1.0.4/bitflags/example_generated/struct.Flags.html#method.from_bits), minus some bits that are hard to parse (I think I commented out all the fields after `<extensions>` tag for example.
